src/utils/image-processor.js

```diff
@@ -61,8 +64,10 @@ const shrinkFile = function(file, settings) {
             df.notify(0.9)
             // console.log('to blob: ' + (new Date() - start))
             if (exif) {
-              op = replaceJpegChunk(blob, 0xe1, [exif.buffer])
+              const newExif = isExifApplied ? setExifOrientation(exif, 1) : exif
+              op = replaceJpegChunk(blob, 0xe1, [newExif.buffer])
```

Comment:
Just suggestion. Actually `setExifOrientation` doesn't create a new copy. It could be shorter.

```suggestion
              if (isExifApplied) {
                setExifOrientation(exif, 1)
              }
              op = replaceJpegChunk(blob, 0xe1, [exif.buffer])
```